tests/check_non_terminating.sh

```diff
@@ -0,0 +1,166 @@
+#!/bin/bash
+set -o nounset
+set -o errexit
+
+DEVFILES_DIR="$(pwd)/stacks"
+
+replaceVariables() {
+    image=$1
+    VAR_KEYS=(liberty-version)
+    VAR_VALUES=(22.0.0.1)
+
+    for i in "${!VAR_KEYS[@]}"; do
+        key='{{' 
+        key+=${VAR_KEYS[i]}
+        key+='}}'
+        value=${VAR_VALUES[i]}
+        image=${image/${key}/${value}} 
+    done
+    echo "$image"
+}
+
+getContainerComponentsNum() {
+    devfilePath=$1
+    component_num=$($YQ_PATH eval '[ .components[] | select(has("container")) ] | length' "$devfilePath" -r)
+    echo "${component_num}"
+}
+
+getName() {
+    devfilePath=$1
+    name=$($YQ_PATH eval '.metadata.name' "$devfilePath" -r)
+    echo "${name}"
+}
+
+getFirstContainerComponentImage() {
+    devfilePath=$1
+
+    image_original=$($YQ_PATH eval '[ .components[] | select(has("container")) ] | .[0].container.image' "$devfilePath" -r)
+    image_processed=$(replaceVariables "${image_original}")
+    echo "${image_processed}"
+}
+
+getFirstContainerComponentCommand() {
+    devfilePath=$1
+    local _gfccc_command=()
+    local _gfccc_command_string=()
+
+    IFS=" " read -r -a _gfccc_command_string <<< "$($YQ_PATH eval '[ .components[] | select(has("container")) ] | .[0].container.command[]? + " "' "$devfilePath" -r | paste -s -d '\0' -)"
+    if (( ${#_gfccc_command_string[@]} == 0 )); then
+        echo ""
+    else
+        for command_word in "${_gfccc_command_string[@]}"; do
+            _gfccc_command+=("${command_word}")
+        done
+        echo "${_gfccc_command[@]}"
+    fi
+}
+
+getFirstContainerComponentArgs() {
+    devfilePath=$1
+    local _gfcca_args=()
+    local _gfcca_args_string=()
+
+    IFS=" " read -r -a _gfcca_args_string <<< "$($YQ_PATH eval '[ .components[] | select(has("container")) ] | .[0].container.args[]? + " "' "$devfilePath" -r | paste -s -d '\0' -)"
+    if (( ${#_gfcca_args_string[@]} == 0 )); then
+      echo ""
+    else
+        for arg in "${_gfcca_args_string[@]}"; do 
+            _gfcca_args+=("${arg}")
+        done
+        echo "${_gfcca_args[@]}"
+    fi
+}
+
+isNonTerminating() {
+    _int_image=$1
+    _int_command=("$2")
+    _int_command_args=("$3")
+
+    namespace=default
+    timeout_in_sec=60 # <== includes image pulling
+
+    echo "  PARAMS: image --> $_int_image, command --> ${_int_command[*]}, args --> ${_int_command_args[*]}"
+ 
+    if [ "${_int_command[*]}" == "null" ] && [ "${_int_command_args[*]}" == "null" ]; then
+        echo "  COMMAND: \"kubectl run test-terminating -n default --attach=false --restart=Never --image=$_int_image\""
+        2>/dev/null 1>/dev/null kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image"
```

Comment:
Instead of using `2>/dev/null 1>/dev/null` it might be slightly cleaner to append `> /dev/null 2>&1`.
```suggestion
        kubectl run test-terminating -n default --attach=false --restart=Never --image="$_int_image" > /dev/null 2>&1
```

Reply:
Makes sense, I will update that.